onnxruntime/test/optimizer/graph_transform_test.cc

```diff
@@ -3846,5 +3847,31 @@ TEST_F(GraphTransformationTests, FilterEnabledOptimizers) {
   ASSERT_TRUE(op_to_count["Add"] == 1);
 }
 
+TEST_F(GraphTransformationTests, PropagateCastOp) {
+  // Any change in the order of the test models will effect the assertions.
```

Comment:
sounds like it's better to zip them into a  std::unordered_map<PathString, int> map?